### PR TITLE
feat(reporting): add flat CSV renderer for the ReportModel

### PR DESCRIPTION
**Pull request by Noah231515**
## What

Adds `internal/reporting/render` — the **first renderer** over the engine's `ReportModel`. `render.CSV(model, groupBy)` produces a flat, **data-only** CSV.

## Layout

Group-by dimensions become leading columns, each detail leaf is one row, and a leading `Row Type` column marks each row `Detail` / `Subtotal` / `Grand Total`:

```csv
Row Type,Paid By,Category,Count,Total,Avg
Detail,Dana,Food,2,300.00,150.00
Detail,Dana,Gas,1,60.00,60.00
Subtotal,Dana,,3,360.00,120.00
Detail,Sam,Food,1,40.00,40.00
Subtotal,Sam,,1,40.00,40.00
Grand Total,,,4,400.00,100.00
```

The `Row Type` column keeps the file safe to aggregate: filter `Row Type=Detail` before summing an additive column (otherwise the subtotal/grand-total rows double-count it), and read the roll-up rows for a non-additive column like `Avg`, which the engine recomputes per level (`120.00 = 360/3`, not a sum or average of the rows above it).

## Design notes

- **Pure downstream consumer.** The renderer fetches and computes nothing and never reaches back into the engine — so a future format or layout touches no upstream code. This is the pluggability the architecture already guarantees via the data-only `ReportModel`, not a new abstraction.
- **`groupBy []render.Dimension`** supplies the dimension order and header labels, because the model carries dimension *keys* but not labels (and an empty report still needs its dimension columns). A future orchestrator builds this from the spec + field catalog.
- **Renders only what the model carries** — subtotal/grand-total rows appear only when the spec toggled them on. It draws no document chrome; per `docs/engine-design.md` §5, CSV is deliberately the minimal renderer. The grouped/visual "looks like the report" layout is the XLSX/PDF renderers' job (separate consumers of the same tree).
- Currency at 2dp, other numbers at full precision, `(None)` buckets use `Meta.NoneLabel`.

## Tests

Models are built via `reporting.Run` and the exact CSV bytes asserted:

- **Sums, aggregates, and non-linear roll-up** — additive columns (Count, Total) roll up (subtotal = sum of its details, grand total = sum of all) while the arithmetic `Avg` is recomputed at each level.
- **Every aggregate function** (SUM / COUNT / AVG / MIN / MAX) at detail, subtotal, and grand-total level.
- **Nested groups** — two levels, subtotal ordering and dimension-column padding.
- **Records mode** with a multi-value label cell (joined, quoted).
- **No grouping**, **totals off** (Row Type column still present), the **`(None)` bucket**, **multi-value fan-out** (the intended double count), an **empty report**, **CSV escaping + unicode** (via a reader round trip), and the per-type/kind **cell formatting** matrix.

No orchestrator or HTTP handler is wired in this slice — the renderer produces bytes and is unit-tested in isolation, consistent with the engine and `ReportDataService` before it.

## Summary by CodeRabbit

* **New Features**
  * Added CSV report export that writes grouped dimension columns plus ordered Detail rows, with Subtotal and Grand Total rows when applicable.
  * Included a leading “Row Type” column to clearly label Detail/Subtotal/Grand Total (kept stable even when grand totals are disabled).
  * Improved formatting and output correctness, including currency precision, label joining, RFC 4180 CSV with CRLF, and consistent “(None)” handling.
* **Documentation**
  * Documented CSV output structure and how renderers consume the generated report model.
* **Tests**
  * Added comprehensive CSV renderer tests covering roll-ups, totals, edge cases, and CSV escaping/unicode round-tripping.

### CODE REVIEW

File: api/internal/reporting/render/csv_test.go [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+	}
+	rows := []reporting.Row{
+		{"paid_by": {reporting.Str("Dana")}, "category": {reporting.Str("Food")}, "amount": {money("100")}},
+	}
+
+	got := mustCSV(t, mustRun(t, spec, catalog, rows), []Dimension{{Key: "paid_by"}})
+	if first, _, _ := strings.Cut(got, "\r\n"); first != "Row Type,paid_by,Category,Total" {
+		t.Errorf("header = %q, want the raw key as the dimension heading", first)
+	}
+}
```
[the remark this answers is not in the document]
Comment (Noah231515) [reply api/internal/reporting/render/csv_test.go:390]: **Accepted — already covered.** `TestColumnHeading_FallsBackToName` (commit `58b8f8e0`) exercises the Label→Name fallback directly, which is why this comment was auto-marked *✅ Addressed in commit 58b8f8e*. I pinned it as a focused unit test rather than through `CSV` because `reporting.Run` always defaults a column's label to its name (`validate.go:258` → `column.go:76`), so the fallback is unreachable via the public API — the direct test is the honest way to cover it.

File: api/internal/reporting/render/csv.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	}
+
+	buffer := new(bytes.Buffer)
+	writer := csv.NewWriter(buffer)
+	writer.UseCRLF = true // RFC 4180 line endings, which Excel prefers.
+	if err := writer.WriteAll(records); err != nil {
+		return nil, err
+	}
+	return buffer.Bytes(), nil
+}
```
[the remark this answers is not in the document]
Comment (Noah231515) [reply api/internal/reporting/render/csv.go:75]: **Accepted, with a correction.** The footgun is real, so I added the depth guard — commit `61b9e60e`. But the proposed `groupDepth`/check rejects a valid **empty grouped report**: an empty report has a childless root, so the derived depth is 0 while `len(groupBy)` is ≥1, and `depth != len(groupBy)` would error on it (breaking `TestCSV_EmptyReportHeaderAndGrandTotalOnly` and `TestCSV_ArithmeticNullRendersBlank`) — which is exactly why `groupBy` is a caller-supplied argument: the tree can't reveal its intended depth when empty. So the guard only fires when the tree actually groups:

```go
if depth := groupDepth(model.Root); depth > 0 && depth != len(groupBy) {
    return nil, fmt.Errorf("render: report groups %d level(s) deep, but %d dimension(s) were supplied", depth, len(groupBy))
}
```

That catches the harmful under-supply case (a report *with* grouped data) without touching empty reports. Added `TestCSV_GroupByDepthMismatchErrors` to cover it.
